api/pkg/controller/cluster/pending.go

```
@@ -202,6 +207,20 @@ func (cc *Controller) ensureCloudProviderIsInitialize(cluster *kubermaticv1.Clus
 	return nil
 }
 
+func (cc *Controller) ensureAutomaticMasterUpdate(c *kubermaticv1.Cluster) error {
+	updateVersion, err := version.BestAutomaticUpdate(c.Spec.MasterVersion, cc.updates)
+	if err != nil {
+		return err
+	}
+
+	if updateVersion != nil {
```

does nil `updateVersion` mean that automatic upgrade is not possible ?

---

`nil` in this case means, there is no update to apply automatically

---

should we return an error is such a case ?

---

No, its fine to not have an automatic update available.
We will run into this case for every cluster which is already on the latest version